snuba/datasets/plans/storage_plan_builder.py

```diff
@@ -176,6 +179,19 @@ def build_and_rank_plans(
         mappers = storage_connection.translation_mappers
         cluster = self.get_cluster(storage, query, settings)
 
+        # Return failure if storage readiness state is not supported in current environment
+        if snuba_settings.READINESS_STATE_FAIL_QUERIES:
+            assert isinstance(storage, ReadableTableStorage)
```

Comment:
I thought you were removing this?

Reply:
Left a comment above. Apparently, we need it to access `get_storage_key()` since storage keys are not part of base.